microsoft_DialoGPT-medium.py

```diff
@@ -0,0 +1,21 @@
+from transformers import AutoModelForCausalLM, AutoTokenizer
+import torch
+
+tokenizer = AutoTokenizer.from_pretrained("microsoft/DialoGPT-medium", padding_side='left')
+model = AutoModelForCausalLM.from_pretrained("microsoft/DialoGPT-medium")
+
+# source: https://huggingface.co/microsoft/DialoGPT-medium
+
+# Let's chat for 5 lines
+for step in range(5):
+    # encode the new user input, add the eos_token and return a tensor in Pytorch
+    new_user_input_ids = tokenizer.encode(input(">> User:") + tokenizer.eos_token, return_tensors='pt')
+
+    # append the new user input tokens to the chat history
+    bot_input_ids = torch.cat([chat_history_ids, new_user_input_ids], dim=-1) if step > 0 else new_user_input_ids
+
+    # generated a response while limiting the total chat history to 1000 tokens, 
+    chat_history_ids = model.generate(bot_input_ids, max_length=1000, pad_token_id=tokenizer.eos_token_id)
+
+    # pretty print last output tokens from bot
+    print("DialoGPT: {}".format(tokenizer.decode(chat_history_ids[:, bot_input_ids.shape[-1]:][0], skip_special_tokens=True)))
```

Comment:
After reviewing all the files, I think some files can be integrated into one as the difference is most of the time only the loaded model. I know that for `experiment/` folder it is more like a playground to quickly implement and understand something, so maybe the review standard should be a bit lower. But since it will go into the master, so I think it is still better to maintain a clean code space everywhere in our major branch. 

Please let me know If it is fine for you to clean and integrate multiple files submitted.